src/plays/shopping-cart/reducer.tsx

```diff
@@ -0,0 +1,66 @@
+import { CartItem, InitialState } from "ShoppingCartTypes";
+
+export default function reducer(state: any, action: any) {
+  switch (action.type) {
+    case "CLEAR_CART":

```

Comment:
It is a best practice to use a constant or a enum for defining the reducer actions. For example in this case, you can have a enum like below
```ts
enum REDUCER_ACTIONS {
    CLEAR_CART = 'CLEAR_CART',
    REMOVE_ITEM = 'REMOVE_ITEM',
    INCREASE_AMOUNT = 'INCREASE_AMOUNT'
}
```
And then use that here in the reducer as well as while dispatching the action.

```suggestion
    case REDUCER_ACTIONS.CLEAR_CART:
```

This helps with avoiding typos and also make your code type safe.